### PR TITLE
[SPARK-40705][SQL] Handle case of using mutable array when converting Row to JSON for Scala 2.13

### PR DESCRIPTION
**Pull request by Amraneze**
### What changes were proposed in this pull request?
I encountered an issue using Spark while reading JSON files based on a schema it throws every time an exception related to conversion of types.

>Note: This issue can be reproduced only with Scala `2.13`, I'm not having this issue with `2.12`

````
Failed to convert value ArraySeq(1, 2, 3) (class of class scala.collection.mutable.ArraySeq$ofRef}) with the type of ArrayType(StringType,true) to JSON.
java.lang.IllegalArgumentException: Failed to convert value ArraySeq(1, 2, 3) (class of class scala.collection.mutable.ArraySeq$ofRef}) with the type of ArrayType(StringType,true) to JSON.
````

If I add ArraySeq to the matching cases, the test that I added passed successfully 
image

With the current code source, the test fails and we have this following error
image

### Why are the changes needed?
If the person is using Scala 2.13, they can't parse an array. Which means they need to fallback to 2.12 to keep the project functioning 

### How was this patch tested?
I added a sample unit test for the case, but I can add more if you want to.

### CODE REVIEW

File: sql/catalyst/src/test/scala/org/apache/spark/sql/RowTest.scala [hunk 1]
```diff
@@ -29,8 +32,8 @@ class RowTest extends AnyFunSpec with Matchers {
 
   val schema = StructType(
     StructField("col1", StringType) ::
-    StructField("col2", StringType) ::
-    StructField("col3", IntegerType) :: Nil)
+      StructField("col2", StringType) ::
```
Comment (Amraneze) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/RowTest.scala:35]: Scalafmt formatted the file, should I revert the changes related to formatting the code ?

Comment (srowen) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/RowTest.scala:35]: Yeah revert all the formatting only changes please

**Review by srowen — COMMENTED** [review]
Looks OK otherwise

File: sql/catalyst/src/test/scala/org/apache/spark/sql/RowTest.scala (hunk 1 above)
Comment (Amraneze) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/RowTest.scala]: I pushed a new commit with the revert of those changes

**Comment by srowen** [comment]
Merged to master/3.3
